crates/ln-dlc-node/src/ln/event_handler.rs

```diff
@@ -692,9 +693,25 @@ where
             "Failed to open channel because maximum channel value exceeded"
         );
 
-        let shadow_channel = Channel::new(0, channel_value, target_node_id);
+        let fake_scid = FakeScid::new(requested_next_hop_scid);
+        let shadow_channel = match self.storage.get_channel_by_fake_scid(fake_scid) {
+            Ok(Some(mut channel)) => {
+                debug_assert!(channel.counterparty == target_node_id);
+                channel.outbound = channel_value;
+                channel
+            }
+
+            Ok(None) => {
+                tracing::warn!(%fake_scid, "Unable to find intercepted channel by fake scid in database, inserting without fake scid instead");
+                Channel::new(0, channel_value, target_node_id, None)
+            }
+            Err(e) => {
+                tracing::error!(%fake_scid, "Failed to load channel by fake scid: {e:#}");
+                Channel::new(0, channel_value, target_node_id, None)
+            }
+        };
```

Comment:
Early in the process, like the scenario below, I think it's OK to fail. But here it would be a shame if we have to fail creation the channel because we don't find the `scid` or there is an error. I would keep this; I don't think it does any harm.